doc/source/text.rst

```diff
@@ -168,28 +168,34 @@ Extracting Substrings
 
 .. _text.extract:
 
-The method ``extract`` (introduced in version 0.13) accepts `regular expressions
-<https://docs.python.org/2/library/re.html>`__ with match groups. Extracting a
-regular expression with one group returns a Series of strings.
+.. versionadded:: 0.13.0
 
-.. ipython:: python
+.. warning::
+
+   In version 0.18.0, ``extract`` gained the ``expand`` argument. When
+   ``expand=False`` (default) it returns a ``Series``, ``Index``, or
+   ``DataFrame``, depending on the subject and regular expression
```

Comment:
> that's exactly right. The purpose of setting it to None is to detect if its explictly set or not. The only valid values are True/False, but we detect None, show the warning, then set it to False.

Not necessarily I think. In the PR of @sinhrks (#10103), he implemented it as the following (copied from the docstring);

```
+    expand : None or bool, default None
+        * If None, return Series/Index (one group) or DataFrame/MultiIndex (multiple groups)
+        * If True, return DataFrame/MultiIndex expanding dimensionality.
+        * If False, return Series/Index.
```

Where both True and False are new behaviours, and `None` indicates the current behaviour